syntax = "proto3";

service API{
	rpc DatosHaciatorre (dataAvion) returns (responseTorre){}
	rpc ComprobarDestino (dataDestino) returns (responseDestino){}
	rpc SolicitarDespegue (dataDespegue) returns (responseDespegue){}
	rpc SolicitarAterrizaje (dataAterrizaje) returns (responseAterrizaje){}
}

message dataAvion{
	string aerolinea = 1;
	string codigo = 2;
	int32 pesoMax = 3;
	int32 CapTanque = 4;
	string ipTorreInicial = 5;
}

message responseTorre {
	bool estado = 1;
	string mensaje = 2;
}

message dataDespegue {
	string destino = 1;
	string aerolinea = 2;
	string codigo = 3;
}

message responseDespegue{
	bool estado = 1;
	string ipdestino = 2;
	string mensaje = 3;
	int32 pista = 4;
	int32 altura = 5;
}

message dataAterrizaje{
	string aerolinea = 1;
	string codigo = 2;
	int32 pesoMax = 3;
	int32 CapTanque = 4;
	string ipTorreActual = 5;
}

message responseAterrizaje{
	bool estado = 1;
	string mensaje = 2;
	int32 puerta = 3;
	int32 pista = 4;
	int32 altura = 5;
}

message dataDestino{
	string destino = 1;
}

message responseDestino{
	bool estado = 1;
}

